superflore/docker.py

```diff
@@ -36,8 +34,19 @@ def map_directory(self, host, container=None, mode='rw'):
     def add_bash_command(self, cmd):
         self.bash_cmds.append(cmd)
 
-    def build(self):
-        self.image = self.client.images.build(path=self.dockerfile_directory)
+    def clear_commands(self):
+        self.bash_cmds = list()
+
+    def build(self, dockerfile):
+        dockerfile_directory = os.path.dirname(dockerfile)
+        if not dockerfile_directory:
```

Comment:
This check doesn't seem right at the moment. It should somehow verify that the Dockefile is a valid dockerfile not that the directory exists.

Reply:
Good point. Fixed.

Reply:
Sorry I meant that you should not just check that it's a directory but check that the Dockerfile itself exists. (aka check the file that is passed exists and then lookup the directory to pass on.) And even better would be to see if the Dockerfile parses/validates but that's extra.